lib/logstash/filters/kv.rb

```diff
@@ -292,8 +301,8 @@ def register
     @remove_char_value_re = Regexp.new("[#{@remove_char_value}]") if @remove_char_value
     @remove_char_key_re = Regexp.new("[#{@remove_char_key}]") if @remove_char_key
 
-    field_split = /[#{@field_split}]/ # TODO: support multi-char splitters by replacing this regexp
-    value_split = /[#{@value_split}]/ # TODO: support multi-char splitters by replacing this regexp
+    field_split = @field_split_multi ? /#{@field_split_multi}/ : /[#{@field_split}]/
```

Comment:
nitpick; since we're using the ivar verbatim, `Regexp::compile(pattern || fallback)` may read more clearly:

~~~ ruby
    field_split = Regexp::compile(@field_split_multi || /[#{@field_split}]/)
    value_split = Regexp::compile(@value_split_multi || /[#{@value_split}]/)
~~~

or, with my other suggestion of `s/multi/pattern/`:
~~~ ruby
    field_split = Regexp::compile(@field_split_pattern || /[#{@field_split}]/)
    value_split = Regexp::compile(@value_split_pattern || /[#{@value_split}]/)
~~~

Reply:
+1